arch/x86/missing_zext.h

```diff
@@ -0,0 +1,22 @@
+#ifndef MISSING_ZEXT_H
+#define MISSING_ZEXT_H
+
+/* Unfortunately GCC _and_ clang didn't support these things until version
+ * 10 and 12, respectively */
+#include <immintrin.h>
+
+#if (defined(__GNUC__) && __GNUC__ < 10) || (defined(__clang__) && __clang__ < 12)
+static inline __m256i _mm256_zextsi128_si256(__m128i a) {
+    __m128i r;
+    volatile __asm__("vmovdqa %1,%0" : "=x" (r) : "x" (_mm256_castsi256_si128(a)));
+    return _mm256_castsi128_si256(r);
+}
+
+static inline __m512i _mm512_zextsi128_si512(__m128i a) {
+    __m128i r;
+    volatile __asm__("vmovdqa %1,%0" : "=x" (r) : "x" (_mm256_castsi512_si128(a)));
+    return _mm512_castsi128_si512(r);
+}
+#endif
+
+#endif
```

Comment:
Maybe include these in `/fallback_builtins.h`, so we have one file for all the workarounds for missing intrinsics.

Reply:
Sure.  These are very specific to x86 and GCC.  Hopefully that's not a problem.